glustercli/cmd/tracing.go

```diff
@@ -0,0 +1,183 @@
+	errTraceUpdateReqFailed = "Failed to update trace configuration"
+)
+
+var (
+	flagTraceJaegerEndpoint       string
+	flagTraceJaegerAgentEndpoint  string
+	flagTraceJaegerSampler        int
+	flagTraceJaegerSampleFraction float64
+)
+
+func init() {
+	// Trace Enable
+	traceEnableCmd.Flags().IntVar(&flagTraceJaegerSampler, "jaeger-sampler", 2, "jaeger sampler (1: Always sample, 2: Probabilistic)")
+	traceEnableCmd.Flags().Float64Var(&flagTraceJaegerSampleFraction, "jaeger-sample-fraction", 0.1, "jaeger sample fraction (min: >0.0, max: <1.0)")
+	// Trace Update
+	traceUpdateCmd.Flags().StringVar(&flagTraceJaegerEndpoint, "jaeger-endpoint", "", "jaeger endpoint")
+	traceUpdateCmd.Flags().StringVar(&flagTraceJaegerAgentEndpoint, "jaeger-agent-endpoint", "", "jaeger agent endpoint")
+	traceUpdateCmd.Flags().IntVar(&flagTraceJaegerSampler, "jaeger-sampler", 2, "jaeger sampler (1: Always sample, 2: Probabilistic)")
+	traceUpdateCmd.Flags().Float64Var(&flagTraceJaegerSampleFraction, "jaeger-sample-fraction", 0.1, "jaeger sample fraction (min: >0.0, max: <1.0)")
+	traceCmd.AddCommand(traceEnableCmd)
+	traceCmd.AddCommand(traceStatusCmd)
+	traceCmd.AddCommand(traceUpdateCmd)
+	traceCmd.AddCommand(traceDisableCmd)
+}
+
+var traceCmd = &cobra.Command{
+	Use:   "trace",
+	Short: helpTraceCmd,
+}
+
+var traceEnableCmd = &cobra.Command{
+	Use:   "enable <jaeger-endpoint> <jaeger-agent-endpoint>",
+	Short: helpTraceEnableCmd,
+	Args:  cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		jaegerEndpoint := args[0]
+		jaegerAgentEndpoint := args[1]
+
+		if jaegerEndpoint == "" || jaegerAgentEndpoint == "" {
+			err := errors.New("One or more Jaeger endpoints not specified")
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+
+		if flagTraceJaegerSampler <= 0 || flagTraceJaegerSampler > 2 {
+			err := errors.New("Invalid Jaeger sampler specified")
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+
+		if flagTraceJaegerSampler == 2 {
+			if flagTraceJaegerSampleFraction <= 0.0 || flagTraceJaegerSampleFraction >= 1.0 {
+				err := errors.New("Invalid Jaeger sample fraction specified")
+				failure(errTraceEnableReqFailed, err, 1)
+			}
+		}
+
+		_, err := client.TraceEnable(tracemgmtapi.SetupTracingReq{
+			JaegerEndpoint:       jaegerEndpoint,
+			JaegerAgentEndpoint:  jaegerAgentEndpoint,
+			JaegerSampler:        flagTraceJaegerSampler,
+			JaegerSampleFraction: flagTraceJaegerSampleFraction,
+		})
+
+		if err != nil {
+			if GlobalFlag.Verbose {
+				log.WithError(err).WithFields(log.Fields{
+					"JaegerEndpoint":       jaegerEndpoint,
+					"JaegerAgentEndpoint":  jaegerAgentEndpoint,
+					"JaegerSampler":        flagTraceJaegerSampler,
+					"JaegerSampleFraction": flagTraceJaegerSampleFraction,
+				}).Error("Trace Enable failed")
+			}
+			failure(errTraceEnableReqFailed, err, 1)
+		}
+		fmt.Println("Trace enable successful")
+	},
+}
+
+func samplerTypeToString(sampler int) string {
```

Comment:
can this be updated as 
```
func samplerTypeToString(type int) string {
sampler:=""	
switch type {
	case 0:
		sampler= "Never"
	case 1:
		sampler= "Always"
	case 2:
		sampler="Probabilistic"
	default:
		sampler="Unknown"
	}
return sampler
}
```

Reply:
I have removed this function and modified the one under pkg/tracing/tracing.go and used that.